ci/upload_audit_events_s3.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+
+from urllib.parse import urljoin
+from botocore.exceptions import ClientError
+
+import json
+import boto3
+import requests
+import os
+import functools
+from datetime import datetime, timedelta, timezone
+
+
+s3_client = boto3.client("s3")
+timestamp_key = "timestamp"
+
+
+class AuditEventsS3Uploader:
+    def __init__(self):
+        self.CF_API_URL = os.environ.get("CF_API_URL")
+        self.UAA_API_URL = os.environ.get("UAA_API_URL")
+        self.UAA_CLIENT_ID = os.environ.get("UAA_CLIENT_ID")
+        self.UAA_CLIENT_SECRET = os.environ.get("UAA_CLIENT_SECRET")
+        self.bucket_name = "{}".format(os.environ["BUCKET"])
+        self.token = self.get_client_credentials_token()
+
+    def get_client_credentials_token(self):
```

Comment:
I don't think it's necessary right now, but something to keep in mind if this becomes more complex or starts taking long enough that it needs to refresh tokens - the cf client library we use in the external-domain-broker-migrator handles token refreshes and auth, so you could have less to think about.